packages/core/src/AttesterSchema.ts

```diff
@@ -0,0 +1,278 @@
+                }
+
+                if (
+                    !this.replDataFields[this.curReplDataFieldCount].addOrPass(
+                        name,
+                        bits
+                    )
+                ) {
+                    if (
+                        this.curReplDataFieldCount + SUM_FIELD_COUNT ==
+                        FIELD_COUNT
+                    ) {
+                        throw Error('Too many replacement and sum fields')
+                    }
+                    this.curReplDataFieldCount++
+
+                    const res = this.replDataFields[
+                        this.curReplDataFieldCount
+                    ].addOrPass(name, bits)
+
+                    if (!res) {
+                        throw Error(
+                            "Couldn't replace " +
+                                name +
+                                ' replacement data field'
+                        )
+                    }
+                }
+            }
+        }
+    }
+
+    getEncodedFieldIndex(entry: string): number {
+        const schema = this.findEntryByName(entry)
+        const updateBy = schema['updateBy']
+
+        if (updateBy == 'sum') {
+            for (let i = 0; i < SUM_FIELD_COUNT; i++) {
+                if (this.sumDataFields[i].field.includes(entry)) {
+                    return i
+                }
+            }
+        } else if (updateBy == 'replace') {
+            for (let i = 0; i < FIELD_COUNT - SUM_FIELD_COUNT; i++) {
+                if (this.replDataFields[i].field.includes(entry)) {
+                    return i
+                }
+            }
+        } else {
+            throw Error('Invalid update type')
+        }
+        return -1
+    }
+
+    update(entry: string, v: bigint) {
+        const schema = this.findEntryByName(entry)
+        const bits = schema['type'].slice(4)
+        const updateBy = schema['updateBy']
+        const maxVal = BigInt(2) ** BigInt(bits) - BigInt(1)
+        const idx = this.getEncodedFieldIndex(entry)
+        if (updateBy == 'sum') {
+            const n = BigInt(this.sumDataFields[idx].getFieldOffset(entry))
+            const val = v + (this.encodedData[idx] >> n)
+
+            if (val >= BigInt(2) ** BigInt(bits)) {
+                throw Error('Summation overflows uint' + bits)
+            }
+
+            this.encodedData[idx] =
+                (this.encodedData[idx] & ~(maxVal << n)) | ((val & maxVal) << n)
+        } else if (updateBy == 'replace') {
+            const n = BigInt(this.replDataFields[idx].getFieldOffset(entry))
+
+            if (v >= BigInt(2) ** BigInt(bits)) {
+                throw Error('Replacement overflows uint' + bits)
+            }
+            this.encodedData[SUM_FIELD_COUNT + idx] =
+                (this.encodedData[SUM_FIELD_COUNT + idx] & ~(maxVal << n)) |
+                ((v & maxVal) << n)
+        } else {
+            throw Error('Entry does not exist in schema')
```

Comment:
this could be tested

Reply:
I realized that the error would never be called since 
```js
const schema = this.findEntryByName(entry)
```
would throw 
```js
throw Error('Entry does not exist in schema')
```
There are now two tests that check for this, one for `update()` and the other for `getEncodedFieldIndex()`